### PR TITLE
Add support for independent qrcode component

### PR DESCRIPTION
**Pull request by devceline**
# Changes
- Export qrcode web component
- Export independent React QRCode component
- Make a `qrcode` page in the labratory to show it off
- Allow optional override for image in `w3m-wallet-image`, this does not affect the public API as this is an internal component
- Allow passing `theme` to `w3m-qrcode` to allow for websites just using the qrcode to be able to pick the theme

### CODE REVIEW

**Review by 0xAsimetriq — COMMENTED** [review]
Glanced over, everything looks good, but I think qrcode is implemented too much like being "it's own library" i.e. initiating ui package / taking in custom theming.

If you need that explicit granular control over this component, I think separate lib would make more sense, but if we can align it more with other components like connect button or network one, can keep it exported.

In essence, users should still init modal as usual and pass options for theme there. Then we just export qrcode component (Let's keep naming aligned and call it `W3mQrcode` or something along those lines). In the end it shouldn't init ui library, not have theme override (instead all of this should be inherited from normal flow).

File: laboratory/src/pages/index.tsx [hunk 1]
```diff
@@ -43,6 +43,12 @@ const cards = [
     description: 'Simple ethereum example that uses WalletConnect Auth SDK',
     link: '/v2Auth',
     color: 'success'
+  },
+  {
```
Comment (0xAsimetriq) [inline laboratory/src/pages/index.tsx:47]: Let's remove this to keep examples to full modal ones with differen't configs, but new component and it's props should be documented in

https://docs.walletconnect.com/2.0/web3modal/react/components
and
https://docs.walletconnect.com/2.0/web3modal/html-js/components

File: packages/react/src/components/QrCode.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export function QrCode(props: JSX.IntrinsicElements['w3m-qrcode']) {
+  const { size } = props
+
+  useEffect(() => {
+    import('@web3modal/ui')
+  })
+
+  return (
+    <div style={{ width: size, height: size }}>
```
Comment (0xAsimetriq) [inline packages/react/src/components/QrCode.tsx:15]: Does display break without the div wrapper, or could we remove it?

File: packages/ui/src/components/w3m-qrcode/index.ts [hunk 3]
```diff
@@ -7,18 +7,20 @@ import { ThemeUtil } from '../../utils/ThemeUtil'
...
+export class W3mQrCode extends LitElement {
   public static styles = [ThemeUtil.globalCss, styles]
 
   // -- state & properties ------------------------------------------- //
   @property() public uri = ''
   @property({ type: Number }) public size = 0
   @property() public imageId? = ''
   @property() public walletId? = ''
+  @property() public imageUrl? = ''
+  @property() public theme?: typeof ThemeCtrl.state.themeMode
```
Comment (0xAsimetriq) [inline packages/ui/src/components/w3m-qrcode/index.ts:19]: Per previous comment, let's inherit theme as before

File: packages/ui/src/components/w3m-qrcode/index.ts [hunk 4]
```diff
@@ -31,11 +33,12 @@ export default class W3mQrCode extends LitElement {
   protected render() {
     return html`
       <div>
-        ${this.walletId
+        ${this.walletId || this.imageUrl
           ? html`
               <w3m-wallet-image
                 walletId=${this.walletId}
                 imageId=${this.imageId}
+                imageUrlOverride=${this.imageUrl}
```
Comment (0xAsimetriq) [inline packages/ui/src/components/w3m-qrcode/index.ts:41]: Maybe this prop on wallet image can remain as `imageUrl` as well?

File: packages/react/src/components/QrCode.tsx (hunk 2 above)
Comment (devceline) [reply packages/react/src/components/QrCode.tsx:15]: It breaks since `w3m-qrcode` takes `width: 100%` and I don't want to break existing usecases for it

**Review by 0xAsimetriq — APPROVED** [review]
Nice, thank you for the changes!

File: laboratory/src/pages/index.tsx (hunk 1 above)
Comment (devceline) [reply laboratory/src/pages/index.tsx]: Will do

**Pull request merged**